anycloud/cli/src/deploy/anycloud_dialoguer.rs

```diff
@@ -0,0 +1,61 @@
+use dialoguer::{theme::ColorfulTheme, Confirm, Input, Select};
+
+pub fn select_with_default(prompt: &str, items: &Vec<String>, default: usize) -> usize {
+  Select::with_theme(&ColorfulTheme::default())
+    .with_prompt(prompt)
+    .items(items)
+    .default(default)
+    .interact()
+    .unwrap()
+}
+
+// pub fn input_with_default_and_validation<F>(prompt: &str, default: String, validation_fn: F) -> String
+// where F: FnMut(&String) -> Result<(), &str> {
+//   Input::with_theme(&ColorfulTheme::default())
+//     .with_prompt(prompt)
+//     .validate_with(validation_fn)
+//     .default(default)
+//     .interact_text()
+//     .unwrap()
+// }
```

Comment:
@cdmistman probably you could help me here. I'm trying to pass a closure to this function so the `.validate_with()` can receive it. As is the code I'm having this error message from the `.validate_with` method:
```
implementation of `std::ops::FnOnce` is not general enough

implementation of `std::ops::FnOnce` is not general enough

note: `std::ops::FnOnce<(&'0 std::string::String,)>` would have to be implemented for the type `F`, for some specific lifetime `'0`...
note: ...but `std::ops::FnOnce<(&std::string::String,)>` is actually implemented for the type `F`rustc 
```
This is the [link](https://docs.rs/dialoguer/0.8.0/dialoguer/struct.Input.html#method.validate_with) to `validate_with` docs.

I've tried different approaches without success.